### DISCUSSION TITLE
패키지 구조 : 도메인형 디렉토리 구조

### DISCUSSION THREAD

**Discussion by Starlight258** [Attica-org/athens-backend#24] [General]
`계층형` 디렉토리 구조보다는 `도메인형` 디렉토리 구조를 채택하여 도메인 간 결합도를 낮추고 응집도를 높이는 것이 어떨까요?
이를 통해 한 패키지에 너무 많은 클래스들이 모이는 것을 방지하고, 도메인별로 나누어 관리함으로써 `OOP`를 더 잘 가져갈 수 있을 것 같아요.

```java
com
 ㄴ example
     ㄴ vivid
         ㄴ domain
         |   ㄴ user
         |   |   ㄴ api
         |   |   ㄴ application
         |   |   ㄴ dao
         |   |   ㄴ domain
         |   |   ㄴ dto
         |   |   ㄴ exception
         |   ㄴ video
         |   |   ㄴ api
         |   |   ㄴ application
         |   |   ㄴ dao
         |   |   ㄴ domain
         |   |   ㄴ dto
         |   |   ㄴ exception
         |   ...
         ㄴ global
             ㄴ auth
             ㄴ common
             ㄴ config
             ㄴ error
             ㄴ infra
             ㄴ util

```

## Reference
https://velog.io/@jsb100800/Spring-boot-directory-package

@big-cir @yybmion

**Comment by big-cir** [comment]
의견에 공감합니다~

**Reply by Starlight258** [reply]
전체적인 패키지 변경은 다음 스프린트에서 하는게 좋을 것 같아요! 코드 충돌을 피하기 위해서요~
새롭게 개발할 부분만 위 패키지 구조를 따라주셔도 좋을 것 같아요.

**Comment by yybmion** [comment]
도메인 개수가 많지 않은 편인 것 같은데 도메인형으로 고려하신 이유가 궁금합니다.

@Starlight258

**Reply by Starlight258** [reply]
도메인도 책임에 따라서 여러 클래스로 나뉠 수 있어요 (srp) 
그래서 한 패키지에 너무 많은 파일이 도메인 구분없이 섞여있으면 파일 찾기가 어렵더라구요 ! ㅠㅠ

**Reply by yybmion** [reply]
그렇군요 감사합니다 !
